qiskit_aqua/_aqua.py

```diff
@@ -116,6 +117,12 @@ def run_algorithm(params, algo_input=None, json_output=False, backend=None):
         else:
             logger.warning("Change basis_gates and coupling_map on a real device is disallowed.")
 
+        use_caching = inputparser.get_section_property(JSONSchema.PROBLEM, 'circuit_caching')
+        if use_caching:
+            caching_naughty_mode = inputparser.get_section_property(JSONSchema.PROBLEM, 'caching_naughty_mode')
```

Comment:
have another name? naughty mode provided limited information. maybe like
- skip_qobj_validation
- reuse_qobj 
?

Reply:
Renamed to caching_reuse_qobj_skip_validation in the schema. Do you want me to break it into two settings?

Reply:
as my understanding, the naughty mode includes the above two features. I think it will be better we have two flags if they are orthogonal.

Reply:
Makes sense - I'll break it up.

Reply:
I think just left `skip_qobj_validation` in the `problem` section.